beacon/prysmgrpc/aggrate.go

```diff
@@ -20,18 +20,19 @@ import (
 )
 
 // GetAggregationData returns aggregation data
-func (b *prysmGRPC) GetAggregationData(ctx context.Context, duty slotqueue.Duty) (*ethpb.AggregateAttestationAndProof, error) {
-	b.waitToSlotTwoThirds(ctx, duty.Duty.AttesterSlot)
+func (b *prysmGRPC) GetAggregationData(ctx context.Context, duty *ethpb.DutiesResponse_Duty, publicKey *bls.PublicKey, shareKey *bls.SecretKey) (*ethpb.AggregateAttestationAndProof, error) {
+	// TODO add aggr to cache in order to prevent duplicate aggr
+	b.waitToSlotTwoThirds(ctx, duty.AttesterSlot)
 
-	slotSig, err := b.signSlot(ctx, duty.Duty.AttesterSlot, duty.ShareSK)
+	slotSig, err := b.signSlot(ctx, duty.AttesterSlot, shareKey)
 	if err != nil {
 		return nil, errors.Wrap(err, "could not sign slot")
 	}
 
 	res, err := b.validatorClient.SubmitAggregateSelectionProof(ctx, &ethpb.AggregateSelectionRequest{
-		Slot:           duty.Duty.AttesterSlot,
-		CommitteeIndex: duty.Duty.CommitteeIndex,
-		PublicKey:      duty.ValidatorPK.Serialize(),
+		Slot:           duty.AttesterSlot,
```

Comment:
@nivBlox let's remove this from this PR and move it for the actual proposal PR

Reply:
in my opinion its better to checkout from this branch with this code inside to reduce additional effort